python-checks/src/main/java/org/sonar/python/checks/BooleanCheckNotInvertedCheck.java

```diff
@@ -46,56 +55,87 @@ private static void checkNotExpression(SubscriptionContext ctx, UnaryExpression
       // Don't raise warning with "not a == b == c" because a == b != c is not equivalent
-      if(!binaryExp.leftOperand().is(Kind.COMPARISON)) {
-        ctx.addIssue(original, String.format(MESSAGE, oppositeOperator(binaryExp.operator())));
+      if (!binaryExp.leftOperand().is(Kind.COMPARISON)) {
+        String oppositeOperator = oppositeOperator(binaryExp.operator());
+        createQuickFix(ctx.addIssue(original, String.format(MESSAGE, oppositeOperator)), oppositeOperator, binaryExp, original);
       }
-    } else if(negatedExpr.is(Kind.IN, Kind.IS) ) {
+    } else if (negatedExpr.is(Kind.IN, Kind.IS)) {
       BinaryExpression isInExpr = (BinaryExpression) negatedExpr;
-      ctx.addIssue(original, String.format(MESSAGE, oppositeOperator(isInExpr.operator(), isInExpr)));
+      String oppositeOperator = oppositeOperator(isInExpr.operator(), isInExpr);
+      createQuickFix(ctx.addIssue(original, String.format(MESSAGE, oppositeOperator)), oppositeOperator, isInExpr, original);
     }
   }
 
-  private static String oppositeOperator(Token operator){
+  private static String oppositeOperator(Token operator) {
     return oppositeOperatorString(operator.value());
   }
 
-  private static String oppositeOperator(Token operator, Expression expr){
+  private static String oppositeOperator(Token operator, Expression expr) {
     String s = operator.value();
-    if(expr.is(Kind.IS) && ((IsExpression) expr).notToken() != null){
+    if (expr.is(Kind.IS) && ((IsExpression) expr).notToken() != null) {
       s = s + " not";
-    } else if(expr.is(Kind.IN) && ((InExpression) expr).notToken() != null){
+    } else if (expr.is(Kind.IN) && ((InExpression) expr).notToken() != null) {
       s = "not " + s;
     }
     return oppositeOperatorString(s);
   }
 
-  static String oppositeOperatorString(String stringOperator){
-    switch (stringOperator){
-      case ">"  :
+  static String oppositeOperatorString(String stringOperator) {
+    switch (stringOperator) {
+      case ">":
         return "<=";
-      case ">=" :
+      case ">=":
         return "<";
-      case "<"  :
+      case "<":
         return ">=";
-      case "<=" :
+      case "<=":
         return ">";
-      case "==" :
+      case "==":
         return "!=";
-      case "!=" :
+      case "!=":
         return "==";
-      case "is" :
+      case "is":
         return "is not";
       case "is not":
         return "is";
-      case "in" :
+      case "in":
         return "not in";
       case "not in":
         return "in";
-      default   :
+      default:
         throw new IllegalArgumentException("Unknown comparison operator : " + stringOperator);
     }
   }
+
+  private static void createQuickFix(PreciseIssue preciseIssue, String oppositeOperator, BinaryExpression toReplace, Tree toRemove) {
+    IssueWithQuickFix issue = (IssueWithQuickFix) preciseIssue;
+
+    PythonTextEdit replaceEdit = null;
+    if (toReplace.is(Kind.IS)) {
```

Comment:
I could imagen a cleaner way to create the `TextEdit` than these nested conditions.